### PR TITLE
Add support of arrow functions in PHP 7.4

### LINKED ISSUE #336: Support PHP 7.4 arrow functions
Hello 👋 !

PHP 7.4 will be released soon and currently arrow functions cause a 

```
message:`ParseError` Code(0) with message "syntax error, unexpected '=>' (T_DOUBLE_ARROW), expecting ')'"
```

To solve this, I propose to duplicate and adapt classic function specs and fixture in the project.
Can this case must be safely added to https://github.com/kahlan/kahlan/blob/master/src/Jit/Parser.php#L85 or do we need to check this apart with the PHP version @jails ? 

It's seems checking `token[0]` is not enough, we must check the if the `T_FUNCTION` is followed by `T_DOUBLE_ARROW` token too.

I can try an implementation this month if needed.

TY

### PR DESCRIPTION
**Pull request by Th3Mouk**
Resolve #336

### CODE REVIEW

File: src/Jit/Parser.php [hunk 1]
```diff
@@ -77,6 +77,7 @@ protected function _parser($content)
         $this->_stream = new TokenStream(['source' => $content, 'wrap' => $this->_states['php']]);
 
         $T_YIELD = defined('HHVM_VERSION') ? 381 : 267;
+        $T_ARROW_FUNCTION = defined('T_FN') ? 343 : 10000;
```
Comment (Th3Mouk) [inline src/Jit/Parser.php]: The usage of the 10000 integer came from this recommendation:
https://www.php.net/manual/en/tokens.php

File: spec/Suite/Jit/Parser.spec.php [hunk 2]
```diff
@@ -61,6 +61,32 @@
...
+        it("parses arrow functions", function () {
+
+            skipIf(defined('HHVM_VERSION') || PHP_VERSION_ID < 70400);
+
+            $sample = file_get_contents('spec/Fixture/Jit/Parser/ArrowFunction.php');
+            $root = Parser::parse($sample);
+            foreach ($root->tree as $node) {
+                if ($node->type === 'function') {
+                    expect($node->name)->toBe('');
+                    expect($node->isClosure)->toBe(true);
```
Comment (samsonasik) [inline spec/Suite/Jit/Parser.spec.php]: toBeTruthy()

File: spec/Suite/Jit/Parser.spec.php [hunk 3]
```diff
@@ -61,6 +61,32 @@
...
+                    expect($node->isMethod)->toBe(false);
```
Comment (samsonasik) [inline spec/Suite/Jit/Parser.spec.php]: toBeFalsy()

**Comment by Th3Mouk** [comment]
@jails @samsonasik 👌

**Comment by jails** [comment]
Awesome :+1:

**Pull request merged**